From 0bacdd1bd8eaea2b1c71279a73c25012e73c986f Mon Sep 17 00:00:00 2001
From: David Wu <david.wu@rock-chips.com>
Date: Wed, 27 Nov 2019 17:19:03 +0800
Subject: [PATCH] ARM: dts: rockchip: Config the pwm2 pinctrl with pull up for
 rk322x

Default pull state of pwm2 and pwm3 pin are up, keep them
when pwm2 and pwm3 are used for pwm regulator. And remove
the pull down config for pwm2 and pwm3, they are not used.

Change-Id: Id8c4767627bd00d224aec734f4a1cacb619c79aa
Signed-off-by: David Wu <david.wu@rock-chips.com>
---
 arch/arm/boot/dts/rk3229-evb.dts | 2 +-
 arch/arm/boot/dts/rk322x.dtsi    | 8 --------
 2 files changed, 1 insertion(+), 9 deletions(-)

diff --git a/arch/arm/boot/dts/rk3229-evb.dts b/arch/arm/boot/dts/rk3229-evb.dts
index 4c034653c941..0f62f0436fc3 100644
--- a/arch/arm/boot/dts/rk3229-evb.dts
+++ b/arch/arm/boot/dts/rk3229-evb.dts
@@ -325,7 +325,7 @@
 &pwm2 {
 	status = "okay";
 	pinctrl-names = "active";
-	pinctrl-0 = <&pwm2_pin_pull_down>;
+	pinctrl-0 = <&pwm2_pin_pull_up>;
 };
 
 &tsadc {
diff --git a/arch/arm/boot/dts/rk322x.dtsi b/arch/arm/boot/dts/rk322x.dtsi
index d254dccb45d3..f52489c1cb47 100644
--- a/arch/arm/boot/dts/rk322x.dtsi
+++ b/arch/arm/boot/dts/rk322x.dtsi
@@ -1592,10 +1592,6 @@
 			pwm2_pin_pull_up: pwm2-pin-pull-up {
 				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_up>;
 			};
-
-			pwm2_pin_pull_down: pwm2-pin-pull-down {
-				rockchip,pins = <1 12 RK_FUNC_2 &pcfg_pull_down>;
-			};
 		};
 
 		pwm3 {
@@ -1606,10 +1602,6 @@
 			pwm3_pin_pull_up: pwm3-pin-pull-up {
 				rockchip,pins = <1 11 RK_FUNC_2 &pcfg_pull_up>;
 			};
-
-			pwm3_pin_pull_down: pwm3-pin-pull-down {
-				rockchip,pins = <1 11 RK_FUNC_2 &pcfg_pull_down>;
-			};
 		};
 
 		spdif {
-- 
2.35.3

